crates/ty_ide/src/completion.rs

```diff
@@ -1247,7 +1247,7 @@ quux.<CURSOR>
         __init_subclass__ :: bound method object.__init_subclass__() -> None
         __module__ :: str
         __ne__ :: bound method object.__ne__(value: object, /) -> bool
-        __new__ :: bound method object.__new__() -> Self
+        __new__ :: bound method object.__new__[Self: object]() -> Self
```

Comment:
I think we should for sure always exclude the upper bound of `Self`, since it's implied by the definition of `Self`. I think probably we should exclude `Self` entirely. (This kind of special handling will probably be easiest to do if we give `Self` its own distinct `TypeVarKind`.)